### DISCUSSION TITLE
Some dynamo BHoM nodes not searchable and add fragment returns null value?...

### DISCUSSION THREAD

**Discussion by JeamesH** [BHoM/BHoM#1361] [Q&A]
Hi there,

Im after some additional help actually running a script in Dynamo. Ive got 2 points I need assistance on.

1 - The Add fragment node has a null list even though Ive fed in 16 true BHoM Floors into it. can someone help explain what is going wrong here?
![image](https://user-images.githubusercontent.com/97482213/152893701-9f901a27-47bd-4ca5-ac70-0964d3926524.png)

Evaluate EPD by volume. When I search for evaluate by volume I can’t find that node?
 
![image](https://user-images.githubusercontent.com/97482213/152893742-130ac660-6f5f-41e8-b56a-ac6c62cd9ff3.png)

 
these 2 issues seem to be stopping the script from running any further?
Once I get past this I was going to go and start on evaluating walls, roofs, ceilings etc.

I would appreciate any help you can assist me with.

Thanks 😊

**Comment by FraserGreenroyd** [comment]
@michaelhoehn are you able to provide any input on this?

**Reply by harley360** [reply]
I believe that @JeamesH is trying to replicate the workflow by @kayleighhoude. 

https://www.autodesk.com/autodesk-university/class/Using-Revit-and-Dynamo-Assess-Embodied-Carbon-2020#video

I've similarly had the values under add fragment return null.
![Screenshot 2022-03-30 173033](https://user-images.githubusercontent.com/102637212/160751715-ae7bb651-c875-49d4-9540-31f95d50be25.jpg)

**Reply by JeamesH** [reply]
That is correct.

I actually had a meeting with Kayleigh earlier this year and somewhat rectified this issue. It does still occur when trying to split families by type.
for example you can see I can easily collect all my collumns and filter them by whether their name contains concrete or metal. As soon as I go to add a fragment the set property value node returns a null value. 

I dont know whether this has to do with the Construction.Layers.Material code block. I think this calls the API field for material but for columns I'm looking for a structural material field within revit?... not sure where to find that
![image](https://user-images.githubusercontent.com/97482213/161167991-f71a1c6c-86a7-40a8-a4ab-2323f73516ee.png)

**Reply by andreea101** [reply]
Hei Jeames, i have had the exact same issue with the fragment node returning null values, how did you solve this?

**Reply by JeamesH** [reply]
Hi Andreea101. I havent solved it yet. ive read the gitbhub front to back and back to front but still cant see why this error is occuring in my script. Even though I can get columns seperated my type name the script wont apply the fragment. this method works absolutely fine for floors so why is it throwing an error with columns? who knows at this stage.

**Reply by oswaldohb** [reply]
Hey, having exactly the same problem @JeamesH @andreea101

**Reply by michaelhoehn** [reply]
@JeamesH @andreea101 @OswaldoHernandezB sorry about all the difficulty with mapping materials. This is actually the core functionality that the next version of the toolkit will hope to solve. 

For this issue, however, I'm fairly confident that this can be resolved by checking the **exact** property name that is being modified for each object type. Currently, BHoM physical objects may actually have different material accessors. For example, a Floor (physical object) can be modified using `Construction.Layers.Material` while a Column can only be modified using `Property.Material`. Currently there is no easy way to distinguish which one you should be using to _set_ with other than exploding the objects (already sorted by type) and tracing through the relevant properties. 

I can't tell what type of objects you are actually working with in your example snippet, but I'd guess they required a different accessor string in order to modify. If switching this doesn't resolve your problem, I'd recommend downloading the latest BHoM 5.1 beta and trying it again as certain material-specific workflows have changed a bit in the past. As I said earlier, we hope to be resolving this issue entirely in the next major LCA toolkit update.

**Reply by JeamesH** [reply]
Hi Michaelhoehn, no need ot apologise the tool kit is coming along nicely and autodeks dont make it easy to get to this level of data through the API. 
thanks for the clarification about material accessors. Flors I have working well. I would have assumed a Coloumn was a BHoM physical object too? you can see in my screenshots below I am using structural columns in revit with the material (concrete, metal etc.) in the family name. In dynamo I am, then pulling columns through BHoM, sorting columns by family name contains noides and then trying to apply EPD data using your 'Property.Material' code. columns are still not picking up those values though.

Ive ben to the BHoM website its still beta version 5.0 that is downloading...
appreciate the work you are all doing on this. thanks.
![image](https://user-images.githubusercontent.com/97482213/175167630-60705f69-c646-4e33-b5fd-98e5c78fa12d.png)
![image](https://user-images.githubusercontent.com/97482213/175167657-3aa9df0a-25d7-4692-b638-776c1d22d389.png)

**Reply by FraserGreenroyd** [reply]
> Ive ben to the BHoM website its still beta version 5.0 that is downloading...

@JeamesH you might need to clear your cache to force it to download 5.1, however, on Monday we'll be releasing the 5.2 beta so it may also be worth waiting till then? 😄

**Reply by JeamesH** [reply]
@FraserGreenroyd thanksfor the tip :) will wait for mondya then. thanks again !
